itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/MetastoreHousekeepingLeaderTestBase.java

```diff
@@ -195,5 +202,30 @@ private void resetThreadStatus() {
     threadNames.forEach((name, status) -> threadNames.put(name, false));
     threadClasses.forEach((thread, status) -> threadClasses.put(thread, false));
   }
+
+  public static class TestLeaderNotification implements LeaderElection.LeadershipStateListener {
```

Comment:
maybe package-private

Reply:
for creating the instance from reflection, https://github.com/apache/hive/pull/6142/files#diff-d697cd0d86def896a111982bc9a1c1d4cdd1175fcc3a21599ce3f0e228613895R109, a public access of the class is needed

Reply:
using reflection in tests is typically not the best approach, as it often indicates underlying design issues